File: HackerRank/Algorithms/3. Strings/String_construction.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <stdbool.h>

int stringConstruction(char* s){
    bool characters[26];
    int i = 0, minimumCost = 0;
    memset(characters, false, 26);
    while (s[i] != '\0') {
        if (!characters[s[i]-'a']) {
            characters[s[i]-'a'] = true;
            minimumCost++;
        }
        i++;
    }
    return minimumCost;
}

int main() {
    int q;
    scanf("%i", &q);
    for(int a0 = 0; a0 < q; a0++){
        char* s = (char *)malloc(512000 * sizeof(char));
        scanf("%s", s);
        int result = stringConstruction(s);
        printf("%d\n", result);
        free(s);
    }
    return 0;
}

File: HackerRank/Algorithms/3. Strings/String_construction.py
#Python3

def stringConstruction(s):
    characters = [False] * 26
    i = 0
    copy_cost = 0
    length = len(s)
    while (i < length):
        index = ord(s[i]) - ord('a')
        if (not characters[index]):
            characters[index] = True
            copy_cost += 1
        i += 1
    return copy_cost

q = int(input().strip())
for a0 in range(q):
    s = input().strip()
    result = stringConstruction(s)
    print(result)
